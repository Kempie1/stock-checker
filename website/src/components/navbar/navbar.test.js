import React from "react"
import Navbar from "./navbar"
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router, Link} from "react-router-dom"


test("<Navbar/> doesn't crash", ()=>{
    const div = document.createElement("div");
    render (<Router><Navbar/></Router>,div)
})

test("<Navbar/> matches snapshot", ()=>{
    const renderedComponent = TestRenderer.create(<Router><Navbar/></Router>)
    expect(renderedComponent).toMatchSnapshot()
})








/*whatButton = () =>{
        if(this.props.user===null)
        {
            return (
        <Link to="/login">
            <Button>Sign In</Button>
        </Link>)
        }
        else{
            return (
                <Link to="/profile">
                    <Button>Profile</Button>
                </Link>
            )
        }
    }*/