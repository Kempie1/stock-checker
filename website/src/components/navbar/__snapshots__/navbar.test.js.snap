// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> matches snapshot 1`] = `
<nav
  className="NavbarItems"
>
  <a
    href="/stock-checker"
    onClick={[Function]}
  >
    <h1
      className="navbar-logo"
    >
      StockChecker
    </h1>
  </a>
  <div
    className="menu-icon"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </div>
  <ul
    className="nav-menu"
  >
    <li
      id="market"
    >
      <a
        className="nav-links"
        href="/stock-checker"
        id="market"
        onClick={[Function]}
      >
        The Market
      </a>
    </li>
    <li
      id="learning"
    >
      <a
        className="nav-links"
        href="/learning"
        id="learning"
        onClick={[Function]}
      >
        Learning
      </a>
    </li>
    <a
      className="mobileButton"
      href="/profile"
      onClick={[Function]}
    >
      <button
        className="btn btn--primary btn--medium"
      >
        Profile
      </button>
    </a>
  </ul>
  <div
    className="search"
  >
    <div>
      <div
        aria-expanded={false}
        aria-label="Dropdown select"
        className="react-dropdown-select undefined css-wmw4vi-ReactDropdownSelect e1gzf2xs0"
        color="#0074D9"
        direction="ltr"
        disabled={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
      >
        <div
          className="react-dropdown-select-content react-dropdown-select-type-single css-v1jrxw-ContentComponent e1gn6jc30"
          onClick={[Function]}
        >
          <input
            className="react-dropdown-select-input css-1njzwg6-InputComponent e11wid6y0"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Select..."
            readOnly={false}
            size={9}
            tabIndex="-1"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <a
    className="desktop"
    href="/profile"
    onClick={[Function]}
  >
    <button
      className="btn btn--primary btn--medium"
    >
      Profile
    </button>
  </a>
</nav>
`;
